linux_os/guide/system/permissions/restrictions/enable_execshield_settings/sysctl_kernel_exec_shield/bash/shared.sh

```diff
@@ -19,7 +19,7 @@ if [ "$(getconf LONG_BIT)" = "64" ] ; then
   if grep --silent noexec {{{ grub2_boot_path }}}/grub*.cfg ; then
         sed -i "s/noexec.*//g" /etc/default/grub
         sed -i "s/noexec.*//g" /etc/grub.d/*
-        GRUBCFG={{{ grub2_boot_path }}}/*.cfg
-        grub2-mkconfig -o "$GRUBCFG"
+        GRUBCFG="{{{ grub2_boot_path }}}/*.cfg"
+        grub2-mkconfig -o $GRUBCFG
```

Comment:
Not sure if this is correct approach.

Reply:
I'd perhaps advocate for:

```
grub2-mkconfig -o "{{{ grub2_boot_path}}}"/grub*.cfg
```

and drop the intermediate variable. 

In practice, `grub2_boot_path` probably won't have spaces but shellcheck/covscan probably doesn't know that and so theoretically it could, hence the quoting, however, the old source is wrong as the quoted glob won't be expanded and instead passed directly to the program (leading it to write out something like `/boot/grub/*.cfg` as a real file on disk).